diff -urN aa/beast-0.6.1/bse/bsecxxutils.h beast-0.6.1/bse/bsecxxutils.h
--- aa/beast-0.6.1/bse/bsecxxutils.h	2004-01-27 16:38:45.000000000 +0100
+++ beast-0.6.1/bse/bsecxxutils.h	2004-07-03 14:36:27.313157120 +0200
@@ -48,8 +48,8 @@
 
 /* --- template errors --- */
 namespace TEMPLATE_ERROR {
-template<typename What, typename Reason> void invalid_type () { TEMPLATE_ERROR::abort; }
-template<typename TYPE> void invalid_argument_type () { TEMPLATE_ERROR::abort; }
+template<typename What, typename Reason> void invalid_type () { abort; }
+template<typename TYPE> void invalid_argument_type () { abort; }
 }
 
 /* --- generally useful templates --- */
