diff -aurN beast-0.5.2-rc2.orig/Makefile.am beast-0.5.2-rc2/Makefile.am
--- beast-0.5.2-rc2.orig/Makefile.am	2003-04-04 18:06:46.000000000 +0200
+++ beast-0.5.2-rc2/Makefile.am	2003-06-21 18:20:04.000000000 +0200
@@ -40,13 +40,13 @@
 # always built PKG_paths.h and create links
 stamp-h: PKG_paths.h
 
-CLEANFILES += PKG_paths.h
+CLEANFILES = PKG_paths.h
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = sfi.pc bse.pc
 $(pkgconfig_DATA): config.status
 
-EXTRA_DIST += $(strip	\
+EXTRA_DIST = $(strip	\
 	TODO		\
 	COPYING.GPL	\
 	COPYING.LGPL	\
diff -aurN beast-0.5.2-rc2.orig/beast-gtk/Makefile.am beast-0.5.2-rc2/beast-gtk/Makefile.am
--- beast-0.5.2-rc2.orig/beast-gtk/Makefile.am	2003-05-25 18:34:27.000000000 +0200
+++ beast-0.5.2-rc2/beast-gtk/Makefile.am	2003-06-21 18:16:07.000000000 +0200
@@ -9,14 +9,14 @@
 # -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_COMPAT_H
 
 # we specify the logging domain here
-DEFS += $(strip \
+DEFS = $(strip \
 	-DG_LOG_DOMAIN=\"BEAST\" \
 )
 
 # special profiling hooks
 PROFILE_SOURCE = # profile.c
-CFLAGS += # $(subst profile.c, -pg -a, $(findstring profile.c, $(<F)))
-LIBS += # -lefence # -pg 
+CFLAGS = # $(subst profile.c, -pg -a, $(findstring profile.c, $(<F)))
+LIBS = # -lefence # -pg 
 
 # link programs against BSE and other neccessary libraries
 progs_LDADD = $(strip \
@@ -93,7 +93,7 @@
 
 
 # gen-sources
-CLEANFILES += xgen-beac xgen-beac xgen-bgth xgen-bgtc xgen-belc xgen-bmh xgen-bmc xgen-bdkd
+CLEANFILES = xgen-beac xgen-beac xgen-bgth xgen-bgtc xgen-belc xgen-bmh xgen-bmc xgen-bdkd
 bstenum_arrays.c: $(beast_headers)
 	(cd ${srcdir} \
 	&& $(GLIB_MKENUMS) \
@@ -182,15 +182,15 @@
 # extra deps
 $(beast_SOURCES) $(bststest_SOURCES) $(tsmview_SOURCES): bstgentypes.h
 
-EXTRA_DIST += $(beast_headers) $(beast_extra_sources) $(beast_extra_files)
-MAINTAINERCLEANFILES += $(beast_built_sources)
-EXTRA_DIST += $(beast_built_sources)
+EXTRA_DIST = $(beast_headers) $(beast_extra_sources) $(beast_extra_files)
+MAINTAINERCLEANFILES = $(beast_built_sources)
+EXTRA_DIST = $(beast_built_sources)
 
 #
 # rules to generate built sources
 #
 gen_source = gen-source.tmp
-CLEANFILES += $(gen_source)
+CLEANFILES = $(gen_source)
 $(beast_sources): $(beast_built_sources)
 
 
diff -aurN beast-0.5.2-rc2.orig/beast-gtk/gxk/Makefile.am beast-0.5.2-rc2/beast-gtk/gxk/Makefile.am
--- beast-0.5.2-rc2.orig/beast-gtk/gxk/Makefile.am	2003-03-31 12:09:33.000000000 +0200
+++ beast-0.5.2-rc2/beast-gtk/gxk/Makefile.am	2003-06-21 18:16:26.000000000 +0200
@@ -7,7 +7,7 @@
 
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) -I$(top_srcdir)/beast-gtk $(GXK_CFLAGS) # -I.. for <gxk/*.h> includes
 
-DEFS += $(strip \
+DEFS = $(strip \
 	-DGTK_DISABLE_COMPAT_H \
 	-DG_LOG_DOMAIN=\"GXK\" \
 )
@@ -90,6 +90,6 @@
 gxktest_SOURCES = gxktest.c
 gxktest_LDADD = $(progs_LDADD)
 
-EXTRA_DIST += $(gxk_headers) $(gxk_extra_sources) $(gxk_built_sources)
+EXTRA_DIST = $(gxk_headers) $(gxk_extra_sources) $(gxk_built_sources)
 MAINTAINERCLEANFILES += $(gxk_built_sources) $(stamp_files)
-CLEANFILES += $(temp_files)
+CLEANFILES = $(temp_files)
diff -aurN beast-0.5.2-rc2.orig/beast-gtk/icons/Makefile.am beast-0.5.2-rc2/beast-gtk/icons/Makefile.am
--- beast-0.5.2-rc2.orig/beast-gtk/icons/Makefile.am	2003-05-06 18:23:24.000000000 +0200
+++ beast-0.5.2-rc2/beast-gtk/icons/Makefile.am	2003-06-21 18:16:35.000000000 +0200
@@ -86,7 +86,7 @@
 @STRIP_END@
 
 gen_sources = xgen-bsgc xgen-bsgh
-CLEANFILES += $(gen_sources) bst-stock-gen.c bst-stock-gen.h
+CLEANFILES = $(gen_sources) bst-stock-gen.c bst-stock-gen.h
 noinst_DATA = bst-stock-gen.c bst-stock-gen.h
 EXTRA_DIST  = $(SRC_IMAGES) $(filter-out STOCK_%, $(STOCK_IMAGE_DEFS))
 
diff -aurN beast-0.5.2-rc2.orig/beast-gtk/images/Makefile.am beast-0.5.2-rc2/beast-gtk/images/Makefile.am
--- beast-0.5.2-rc2.orig/beast-gtk/images/Makefile.am	2003-04-13 17:09:13.000000000 +0200
+++ beast-0.5.2-rc2/beast-gtk/images/Makefile.am	2003-06-21 18:16:49.000000000 +0200
@@ -15,10 +15,10 @@
 
 imagesdir = ${beastimagesdir}
 images_DATA = ${public_images}
-EXTRA_DIST += ${public_images} ${private_images}
+EXTRA_DIST = ${public_images} ${private_images}
 
 gen_sources           = xgen-bstih
-CLEANFILES           += $(gen_sources) beast-images.h
+CLEANFILES            = $(gen_sources) beast-images.h
 noinst_DATA           = beast-images.h
 
 beast-images.h:	Makefile ${public_images}
diff -aurN beast-0.5.2-rc2.orig/bse/Makefile.am beast-0.5.2-rc2/bse/Makefile.am
--- beast-0.5.2-rc2.orig/bse/Makefile.am	2003-06-08 17:57:29.000000000 +0200
+++ beast-0.5.2-rc2/bse/Makefile.am	2003-06-21 18:18:01.000000000 +0200
@@ -18,7 +18,7 @@
 # due to automake-1.3 not featuring target specific defines,
 # the G_LOG_DOMAIN needs to be special cased, depending on
 # the source file sets ;(
-DEFS += $(strip \
+DEFS = $(strip \
 	$(patsubst %, -DG_LOG_DOMAIN=\"BSE\" -DBSE_COMPILATION, \
 	              $(filter $(<F), $(bse_c_sources))) \
 	$(patsubst %, -DBSE_BUILTIN_NAME=$(strip $(subst -,_, $(subst .,_,$(basename $(<F))))) \
@@ -183,10 +183,10 @@
 bse_h_sources = $(bse_private_h_sources) $(bse_public_h_sources) $(bse_built_h_sources)
 libbseinclude_HEADERS = $(bse_public_h_sources) $(bse_built_h_sources)
 libbse_la_SOURCES = $(bse_c_sources) $(bse_builtin_c_sources)
-MAINTAINERCLEANFILES += $(bse_built_h_sources) $(bse_built_c_sources)
-EXTRA_HEADERS +=
-EXTRA_DIST += $(bse_private_h_sources) $(bse_proc_sources)
-EXTRA_DIST += $(bse_built_c_sources) $(bse_built_h_sources) $(bse_extra_sources)
+MAINTAINERCLEANFILES = $(bse_built_h_sources) $(bse_built_c_sources)
+EXTRA_HEADERS =
+EXTRA_DIST = $(bse_private_h_sources) $(bse_proc_sources)
+EXTRA_DIST = $(bse_built_c_sources) $(bse_built_h_sources) $(bse_extra_sources)
 GLIB_MKENUMS = glib-mkenums
 SFIDL = ../sfi/sfidl	# WATCH OUT: $(SFIDL) works only for srcdir==builddir with cd $(srcdir)
 
@@ -197,7 +197,7 @@
 # setup autogeneration dependancies
 tmp_files = xgen-sbgth xgen-bgtc xgen-bgtac xgen-sbeac xgen-belc xgen-bbec xgen-bbac \
 	    xgen-bnac xgen-bmx xgen-bmh xgen-bmc xgen-bgasi
-CLEANFILES += $(tmp_files) @DVLBUILD@ $(bse_proc_gen_sources)
+CLEANFILES = $(tmp_files) @DVLBUILD@ $(bse_proc_gen_sources)
 $(OBJECTS): $(bse_built_h_sources)
 # piggyback generation through sideeffects, needs NOP-rule for dep-tracking, must be in srcdir
 $(srcdir)/bsegentypes.h: @DVLBUILD@ stamp-bsegentypes.h ;
@@ -272,7 +272,7 @@
 	&& ./bseprocidl >> xgen-bgasi \
 	&& (cmp -s xgen-bgasi bse.sfidl || cp xgen-bgasi bse.sfidl) \
 	&& rm -f xgen-bgasi
-CLEANFILES += bse.sfidl
+CLEANFILES = bse.sfidl
 noinst_DATA = bse.sfidl
 
 
diff -aurN beast-0.5.2-rc2.orig/bse/icons/Makefile.am beast-0.5.2-rc2/bse/icons/Makefile.am
--- beast-0.5.2-rc2.orig/bse/icons/Makefile.am	2003-03-10 05:40:35.000000000 +0100
+++ beast-0.5.2-rc2/bse/icons/Makefile.am	2003-06-21 18:17:23.000000000 +0200
@@ -11,11 +11,11 @@
 	cd $(srcdir) \
 	&& echo "## Generated data (by make $@)" > $@ \
 	&& for f in $(icon_pattern) ; do \
-	    echo "icon_list += $$f" >>$@ ; \
+	    echo "icon_list = $$f" >>$@ ; \
 	done \
 	&& echo "## Generated data ends here" >> $@
 
-EXTRA_DIST += $(strip \
+EXTRA_DIST = $(strip \
 	$(icon_list) \
 )
 
diff -aurN beast-0.5.2-rc2.orig/bse/tests/Makefile.am beast-0.5.2-rc2/bse/tests/Makefile.am
--- beast-0.5.2-rc2.orig/bse/tests/Makefile.am	2003-05-20 18:17:40.000000000 +0200
+++ beast-0.5.2-rc2/bse/tests/Makefile.am	2003-06-21 18:17:34.000000000 +0200
@@ -5,9 +5,9 @@
 
 SUBDIRS =
 
-INCLUDES += -I$(top_srcdir) -I$(top_builddir) $(SFI_CFLAGS)
+INCLUDES = -I$(top_srcdir) -I$(top_builddir) $(SFI_CFLAGS)
 
-DEFS += -DG_LOG_DOMAIN=\"BSE-TEST\" # -DG_DISABLE_CONST_RETURNS
+DEFS = -DG_LOG_DOMAIN=\"BSE-TEST\" # -DG_DISABLE_CONST_RETURNS
 
 
 TESTS = testwavechunk testfft
@@ -26,5 +26,5 @@
 mathtests_LDADD = $(progs_ldadd)
 
 
-EXTRA_DIST += $(strip \
+EXTRA_DIST = $(strip \
 )
diff -aurN beast-0.5.2-rc2.orig/bse/zintern/Makefile.am beast-0.5.2-rc2/bse/zintern/Makefile.am
--- beast-0.5.2-rc2.orig/bse/zintern/Makefile.am	2003-04-28 00:18:28.000000000 +0200
+++ beast-0.5.2-rc2/bse/zintern/Makefile.am	2003-06-21 18:17:43.000000000 +0200
@@ -10,7 +10,7 @@
 )
 
 gen_sources = xgen-bzc xgen-bzh
-CLEANFILES += $(gen_sources) bse-zfile.c bse-zfile.h
+CLEANFILES  = $(gen_sources) bse-zfile.c bse-zfile.h
 noinst_DATA = bse-zfile.c # bse-zfile.h
 EXTRA_DIST  = $(filter-out BSE_%, $(ZFILE_DEFS))
 
diff -aurN beast-0.5.2-rc2.orig/bsw/Makefile.am beast-0.5.2-rc2/bsw/Makefile.am
--- beast-0.5.2-rc2.orig/bsw/Makefile.am	2003-03-10 10:03:15.000000000 +0100
+++ beast-0.5.2-rc2/bsw/Makefile.am	2003-06-21 18:18:19.000000000 +0200
@@ -15,7 +15,7 @@
 # due to automake-1.3 not featuring target specific defines,
 # the G_LOG_DOMAIN needs to be special cased, depending on
 # the source file sets ;(
-DEFS += @STRIP_BEGIN@ \
+DEFS = @STRIP_BEGIN@ \
 	-DG_LOG_DOMAIN=\"BSW\" \
 @STRIP_END@
 
@@ -66,10 +66,10 @@
 bsw_h_sources = $(bsw_private_h_sources) $(bsw_public_h_sources) $(bsw_built_public_sources)
 libbswinclude_HEADERS = $(bsw_public_h_sources) $(bsw_built_public_sources)
 libbsw_la_SOURCES = $(bsw_c_sources)
-CLEANFILES += $(bsw_built_public_sources) $(bsw_built_sources)
-EXTRA_HEADERS +=
-EXTRA_DIST += $(bsw_private_h_sources)
-EXTRA_DIST += $(bsw_built_sources) $(bsw_built_public_sources) $(bsw_extra_sources)
+CLEANFILES = $(bsw_built_public_sources) $(bsw_built_sources)
+EXTRA_HEADERS =
+EXTRA_DIST = $(bsw_private_h_sources)
+EXTRA_DIST = $(bsw_built_sources) $(bsw_built_public_sources) $(bsw_extra_sources)
 
 SFIDL = $(top_builddir)/sfi/sfidl
 
@@ -78,7 +78,7 @@
 #
 # setup autogeneration dependancies
 gen_sources = xgen-sbpah xgen-bpac
-CLEANFILES += $(gen_sources)
+CLEANFILES = $(gen_sources)
 # initial creation of the real stamp-* files
 bswgenapi.h: stamp-bswgenapi.h ;
 # normal autogeneration rules
@@ -133,4 +133,4 @@
 #
 # auxillary files
 #
-EXTRA_DIST += #	TODO
+EXTRA_DIST = #	TODO
diff -aurN beast-0.5.2-rc2.orig/configure.in beast-0.5.2-rc2/configure.in
--- beast-0.5.2-rc2.orig/configure.in	2003-06-20 00:23:24.000000000 +0200
+++ beast-0.5.2-rc2/configure.in	2003-06-21 18:23:41.000000000 +0200
@@ -13,13 +13,13 @@
 #
 # BSE version numbers
 #
-AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
+#AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
 BSE_MAJOR_VERSION=0
 BSE_MINOR_VERSION=5
 BSE_MICRO_VERSION=2
 BSE_INTERFACE_AGE=0
 BSE_BINARY_AGE=0
-AC_DIVERT_POP()dnl
+#AC_DIVERT_POP()dnl
 dnl#
 dnl# Making releases:
 dnl#   BSE_MICRO_VERSION += 1;
@@ -57,13 +57,13 @@
 
 dnl figure debugging default, prior to $ac_help setup
 dnl
-AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
+#AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
 if test `expr $BSE_MINOR_VERSION \% 2 \| $BSE_MAJOR_VERSION \< 1` = 1 ; then
         debug_default=yes
 else
         debug_default=minimum
 fi
-AC_DIVERT_POP()dnl
+#AC_DIVERT_POP()dnl
 
 
 # initialize automake, don't define PACKAGE and VERSION
diff -aurN beast-0.5.2-rc2.orig/docs/generated/Makefile.am beast-0.5.2-rc2/docs/generated/Makefile.am
--- beast-0.5.2-rc2.orig/docs/generated/Makefile.am	2003-05-11 03:32:09.000000000 +0200
+++ beast-0.5.2-rc2/docs/generated/Makefile.am	2003-06-21 18:18:39.000000000 +0200
@@ -48,7 +48,7 @@
 
 # extra dist
 build_tools = scandocs.pl
-EXTRA_DIST += ${build_tools}
+EXTRA_DIST = ${build_tools}
 
 #
 # reference docu generation
@@ -173,8 +173,8 @@
 	$(distdocs)					\
 )
 endif
-CLEANFILES += $(conditional_clean)
-EXTRA_DIST += $(distdocs)
+CLEANFILES = $(conditional_clean)
+EXTRA_DIST = $(distdocs)
 
 #
 # website generation
@@ -229,7 +229,7 @@
 	tar zxpf  beast-site-nav.tar.gz  -C $(BEAST_PREVIEW)
 site-update: site-update-con
 site-upgrade: site-update-con site-update-img site-update-nav
-CLEANFILES += beast-site-*.tar.gz beast-site-*.tar.bz2 index.html
+CLEANFILES = beast-site-*.tar.gz beast-site-*.tar.bz2 index.html
 
 
 #
@@ -247,7 +247,7 @@
 	  xmllint --noout --loaddtd --valid $$i ; \
 	done
 
-MAINTAINERCLEANFILES +=
+MAINTAINERCLEANFILES =
 
 # intermediate files we don't need to keep
 .INTERMEDIATE: $(strip \
diff -aurN beast-0.5.2-rc2.orig/plugins/Makefile.am beast-0.5.2-rc2/plugins/Makefile.am
--- beast-0.5.2-rc2.orig/plugins/Makefile.am	2003-05-10 18:14:10.000000000 +0200
+++ beast-0.5.2-rc2/plugins/Makefile.am	2003-06-21 18:19:01.000000000 +0200
@@ -5,9 +5,9 @@
 
 SUBDIRS = icons . freeverb
 
-CFLAGS   += $(EXTRA_CFLAGS)
-INCLUDES += -I$(top_srcdir) -I. $(BSE_CFLAGS) -DG_LOG_DOMAIN=BSE_PLUGIN_NAME
-INCLUDES += -DBSE_PLUGIN_FALLBACK="\"$(strip $(basename $(<F)))\""
+CFLAGS   = $(EXTRA_CFLAGS)
+INCLUDES = -I$(top_srcdir) -I. $(BSE_CFLAGS) -DG_LOG_DOMAIN=BSE_PLUGIN_NAME
+INCLUDES = -DBSE_PLUGIN_FALLBACK="\"$(strip $(basename $(<F)))\""
 GLIB_MKENUMS = glib-mkenums
 
 # install plugins under:
@@ -45,7 +45,7 @@
 Makefile.plugins: $(srcdir)/listplugins.sh $(PLUGIN_CFILES) Makefile.am
 	$(srcdir)/listplugins.sh $(PLUGIN_CFILES) >$@
 
-EXTRA_DIST += $(strip \
+EXTRA_DIST = $(strip \
         $(EXTRA_HEADERS) \
         $(EXTRA_CFILES) \
         $(plugins_built_sources) \
@@ -75,7 +75,7 @@
 	    $< >> xgen-enums-$(@F) \
 	&& cp xgen-enums-$(@F) $(@F) \
 	&& rm -f xgen-enums-$(@F)
-CLEANFILES += $(plugins_built_sources) xgen-enums-*
+CLEANFILES = $(plugins_built_sources) xgen-enums-*
 
 ## include generated plugin Makefile
 include Makefile.plugins
diff -aurN beast-0.5.2-rc2.orig/plugins/freeverb/Makefile.am beast-0.5.2-rc2/plugins/freeverb/Makefile.am
--- beast-0.5.2-rc2.orig/plugins/freeverb/Makefile.am	2003-03-10 05:41:12.000000000 +0100
+++ beast-0.5.2-rc2/plugins/freeverb/Makefile.am	2003-06-21 18:19:11.000000000 +0200
@@ -1,10 +1,10 @@
 # Freeverb Makefile
 #
 
-CFLAGS   += $(EXTRA_CFLAGS)
-CPPFLAGS += $(EXTRA_CXXFLAGS)
-INCLUDES += -I$(top_srcdir) -I. $(BSE_CFLAGS) -DG_LOG_DOMAIN=BSE_PLUGIN_NAME
-INCLUDES += -DBSE_PLUGIN_FALLBACK="\"$(strip $(basename $(<F)))\""
+CFLAGS   = $(EXTRA_CFLAGS)
+CPPFLAGS = $(EXTRA_CXXFLAGS)
+INCLUDES = -I$(top_srcdir) -I. $(BSE_CFLAGS) -DG_LOG_DOMAIN=BSE_PLUGIN_NAME
+INCLUDES = -DBSE_PLUGIN_FALLBACK="\"$(strip $(basename $(<F)))\""
 
 # install plugins under:
 plugindir = $(bseplugindir)
@@ -18,7 +18,7 @@
 
 plugin_LTLIBRARIES = freeverb.la
 
-EXTRA_DIST += $(strip \
+EXTRA_DIST = $(strip \
 	bsefreeverb.h \
 	bsefreeverbcpp.h \
 	denormals.h \
diff -aurN beast-0.5.2-rc2.orig/plugins/icons/Makefile.am beast-0.5.2-rc2/plugins/icons/Makefile.am
--- beast-0.5.2-rc2.orig/plugins/icons/Makefile.am	2003-03-10 05:41:12.000000000 +0100
+++ beast-0.5.2-rc2/plugins/icons/Makefile.am	2003-06-21 18:19:18.000000000 +0200
@@ -11,11 +11,11 @@
 	cd $(srcdir) \
 	&& echo "## Generated data (by make $@)" > $@ \
 	&& for f in $(icon_pattern) ; do \
-	    echo "icon_list += $$f" >>$@ ; \
+	    echo "icon_list = $$f" >>$@ ; \
 	done \
 	&& echo "## Generated data ends here" >> $@
 
-EXTRA_DIST += $(strip \
+EXTRA_DIST = $(strip \
 	$(icon_list) \
 )
 
diff -aurN beast-0.5.2-rc2.orig/sfi/Makefile.am beast-0.5.2-rc2/sfi/Makefile.am
--- beast-0.5.2-rc2.orig/sfi/Makefile.am	2003-05-20 18:24:41.000000000 +0200
+++ beast-0.5.2-rc2/sfi/Makefile.am	2003-06-21 18:19:43.000000000 +0200
@@ -5,13 +5,13 @@
 
 SUBDIRS =
 
-INCLUDES += -I$(top_srcdir) -I$(top_builddir) $(SFI_CFLAGS)
+INCLUDES = -I$(top_srcdir) -I$(top_builddir) $(SFI_CFLAGS)
 
 # supported defines: SFI_DISABLE_DEBUG
 
 # public libraries
 lib_LTLIBRARIES = libsfi.la
-DEFS += -DG_LOG_DOMAIN=\"SFI\" -DG_DISABLE_CONST_RETURNS
+DEFS = -DG_LOG_DOMAIN=\"SFI\" -DG_DISABLE_CONST_RETURNS
 libsfiincludedir = $(includedir)/sfi
 libsfi_la_LDFLAGS = $(strip \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
@@ -54,15 +54,15 @@
 #
 SFIDL = $(top_builddir)/sfi/sfidl
 temp_files =  xgen-tih xgen-tic xgen-sch
-CLEANFILES += $(temp_files)
-CLEANFILES += testidl.h testidl.c
+CLEANFILES = $(temp_files)
+CLEANFILES = testidl.h testidl.c
 testidl.h: $(srcdir)/testidl.sfidl sfidl
 	$(SFIDL) --header --implementation --boxed $(srcdir)/testidl.sfidl > xgen-tih \
 	&& mv xgen-tih testidl.h
 testidl.c: $(srcdir)/testidl.sfidl sfidl
 	$(SFIDL) --source --implementation --boxed --init test_types_init $(srcdir)/testidl.sfidl > xgen-tic \
 	&& mv xgen-tic testidl.c
-EXTRA_DIST += testidl.sfidl
+EXTRA_DIST = testidl.sfidl
 
 sficonfig.h: $(top_builddir)/config.status Makefile
 	echo "/* Generated data from $< (by make $@) */" > xgen-sch \
@@ -76,7 +76,7 @@
 	&& echo "/* Generated data ends here */" >> xgen-sch \
 	&& (cmp -s xgen-sch $@ || cp xgen-sch $@) \
 	&& rm -f xgen-sch
-CLEANFILES += sficonfig.h xgen-sch
+CLEANFILES = sficonfig.h xgen-sch
 $(srcdir)/sfithreads.c: sficonfig.h
 $(OBJECTS): sficonfig.h
 
@@ -86,8 +86,8 @@
 noinst_PROGRAMS = sfidl testsfi
 # source files
 sfidl_SOURCES = sfidl.cc sfidl-namespace.cc sfidl-options.cc sfidl-parser.cc glib-extra.c
-EXTRA_DIST += sfidl-namespace.h sfidl-options.h sfidl-parser.h
-progs_extra_sources += sfidl-namespace.h
+EXTRA_DIST = sfidl-namespace.h sfidl-options.h sfidl-parser.h
+progs_extra_sources = sfidl-namespace.h
 sfidl_LDADD = $(SFI_LIBS) -lm
 testsfi_SOURCES = testsfi.c
 testsfi_LDADD = libsfi.la
@@ -103,8 +103,8 @@
 #
 TOYPROF_H_SOURCES = toyprof.h toyprof-mem.h
 TOYPROF_C_SOURCES = toyprof.c toyprof-mem.c
-EXTRA_DIST += toyprof.pl toyprof.README $(TOYPROF_H_SOURCES)
+EXTRA_DIST = toyprof.pl toyprof.README $(TOYPROF_H_SOURCES)
 
 
-EXTRA_DIST += $(sfi_public_headers) $(sfi_extra_sources) $(progs_extra_sources) $(sfi_built_sources)
-MAINTAINERCLEANFILES += $(sfi_built_sources)
+EXTRA_DIST = $(sfi_public_headers) $(sfi_extra_sources) $(progs_extra_sources) $(sfi_built_sources)
+MAINTAINERCLEANFILES = $(sfi_built_sources)
diff -aurN beast-0.5.2-rc2.orig/shell/Makefile.am beast-0.5.2-rc2/shell/Makefile.am
--- beast-0.5.2-rc2.orig/shell/Makefile.am	2003-06-17 04:29:58.000000000 +0200
+++ beast-0.5.2-rc2/shell/Makefile.am	2003-06-21 18:19:54.000000000 +0200
@@ -9,11 +9,11 @@
 # built sources
 #
 built_sources = # bsescm-genglue.c
-MAINTAINERCLEANFILES += $(built_sources)
-EXTRA_DIST += $(built_sources)
+MAINTAINERCLEANFILES = $(built_sources)
+EXTRA_DIST = $(built_sources)
 # temporary build files
 gen_sources = xgen-bsggc
-CLEANFILES += $(gen_sources)
+CLEANFILES = $(gen_sources)
 
 #
 # programs to build
@@ -25,7 +25,7 @@
 bsesh_SOURCES = bsesh.c bsescminterp.c cxxdummy.cc
 bsesh_LDADD = $(programs_ldadd) $(BSE_SHELL_LIBS)
 bsescminterp.c: # bsescm-genglue.c
-EXTRA_DIST += bsescminterp.h
+EXTRA_DIST = bsescminterp.h
 
 
 #
@@ -38,4 +38,4 @@
 	progressor-example.scm	\
 	modules2grid.scm	\
 )
-EXTRA_DIST += $(public_scripts) test.scm
+EXTRA_DIST = $(public_scripts) test.scm
